CppCustomVisualizer2/dll/RootVisualizer.cpp

```diff
@@ -0,0 +1,376 @@
+            i,
+            &pEvaluationResult
+        );
+        if (FAILED(hr))
+        {
+            return hr;
+        }
+
+        CComPtr<DkmChildVisualizedExpression> pChildVisualizedExpression;
+        hr = DkmChildVisualizedExpression::Create(
+            m_pVisualizedExpression->InspectionContext(),
+            m_pVisualizedExpression->VisualizerId(),
+            m_pVisualizedExpression->SourceId(),
+            m_pVisualizedExpression->StackFrame(),
+            pPointerValueHome,
+            pEvaluationResult,
+            m_pVisualizedExpression,
+            i,
+            pChildVisualizer,
+            &pChildVisualizedExpression
+        );
+        if (FAILED(hr))
+        {
+            return hr;
+        }
+
+        childItems.AddTail(pChildVisualizedExpression);
+    }
+
+    CAutoDkmArray<DkmChildVisualizedExpression*> resultValues;
+    DkmAllocArray(childItems.GetCount(), &resultValues);
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+
+    UINT32 j = 0;
+    POSITION pos = childItems.GetHeadPosition();
+    while (pos != NULL)
+    {
+        CComPtr<DkmChildVisualizedExpression> pCurr = childItems.GetNext(pos);
+        resultValues.Members[j] = pCurr.Detach();
+        j++;
+    }
+
+    *pItems = resultValues.Detach();
+
+    return hr;
+}
+
+HRESULT CRootVisualizer::GetSize(
+    _In_ Evaluation::DkmVisualizedExpression* pVisualizedExpression,
+    _In_ DkmString* pFullName,
+    _In_ LPCWSTR pMemberName,
+    _In_ bool rootIsPointer,
+    _Out_ unsigned long long* pSize
+)
+{
+    HRESULT hr = S_OK;
+
+    CString evalText;
+    if (rootIsPointer)
+    {
+        evalText.Format(L"(%s)->%s.size()", pFullName->Value(), pMemberName);
+    }
+    else
+    {
+        evalText.Format(L"(%s).%s.size()", pFullName->Value(), pMemberName);
+    }
+    CComPtr<DkmString> pEvalText;
+    hr = DkmString::Create(DkmSourceString(evalText), &pEvalText);
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+
+    CComPtr<DkmLanguageExpression> pLanguageExpression;
+    hr = DkmLanguageExpression::Create(
+        pVisualizedExpression->InspectionContext()->Language(),
+        DkmEvaluationFlags::TreatAsExpression,
+        pEvalText,
+        DkmDataItem::Null(),
+        &pLanguageExpression
+    );
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+    CComPtr<DkmEvaluationResult> pEvalResult;
+    hr = pVisualizedExpression->EvaluateExpressionCallback(
+        pVisualizedExpression->InspectionContext(),
+        pLanguageExpression,
+        pVisualizedExpression->StackFrame(),
+        &pEvalResult
+    );
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+
+    CComPtr<DkmSuccessEvaluationResult> pSuccessEvalResult = DkmSuccessEvaluationResult::TryCast(pEvalResult);
+    if (pSuccessEvalResult == nullptr || pSuccessEvalResult->Value() == nullptr)
+    {
+        return E_FAIL;
+    }
+
+    *pSize = wcstoull(pSuccessEvalResult->Value()->Value(), nullptr, 10);
```

Comment:
>10 [](http://example.com/codeflow?start=69&length=2)

Base should be '0' so that `wcstoull` will figure it out based on the `0x` prefix.